### DISCUSSION TITLE
Why there's no AsyncRelayCommand attribute

### DISCUSSION THREAD

**Discussion by msnoshain** [CommunityToolkit/dotnet#388] [Q&A] [answered]
I prefer to use attributes to simplify my code:

`[RelayCommand]`
`void TryLoginAndNavigate()`
`{`

` }`

But I found that there's no AsyncRelayCommand attribute.
Is RelayCommand attribute able to handle async functions?

**Comment by Arlodotexe** [comment, marked as answer]
The `[RelayCommand]` attribute will generate an `AsyncRelayCommand` if the method it's attached to is asynchronous and returns a `Task`.

**Reply by msnoshain** [reply]
Thanks! I wrote a function that returns void by misktake. It's no wonder the source generater kept generating RelayCommand for me.
